src/test/java/org/corfudb/runtime/clients/TestClientRouter.java

```diff
@@ -152,6 +158,7 @@ public IClientRouter addClient(IClient client) {
         // Set the message fields.
         message.setClientID(clientID);
         message.setRequestID(thisRequest);
+        message.setEpoch(getEpoch());
```

Comment:
Should be but the TestClientRouter code is separate from NettyClientRouter. We need to merge the TestRouter and NettyRouter code at some point so that the behavior stays the same. Not sure if it is completely possible but will have to attempt it soon.